kfs-core/src/main/java/edu/arizona/kfs/gl/businessobject/GecEntry.java

```diff
@@ -0,0 +1,38 @@
+package edu.arizona.kfs.gl.businessobject;
+
+import org.kuali.kfs.gl.businessobject.Entry;
+
+import edu.arizona.kfs.gl.businessobject.lookup.GecEntryHelperServiceImpl;
+
+/**
+ * This class is used to retrieve the data in the table for Entry objects, add a the
+ * gecDocumentNumber field, and provide dynamically generated objectIds since they do
+ * not exist in the database for use in GEC Lookups.
+ *
+ * @author Adam Kost <kosta@email.arizona.edu> with some code adapted from UCI
+ */
+
+public class GecEntry extends Entry {
+
+    private static final long serialVersionUID = 7915615641174612210L;
+
+    // transient, dynamically generated fields
+    private transient String gecDocumentNumber;
+
+    public String getGecDocumentNumber() {
+        return gecDocumentNumber;
+    }
+
+    public void setGecDocumentNumber(String gecDocumentNumber) {
+        this.gecDocumentNumber = gecDocumentNumber;
+    }
+
+    public String getObjectId() {

```

Comment:
I think using objectId is less than ideal for several reasons. First, is you're overloading this field beyond just a UUID, and in a way that will be counterintuitive to the rest of the entire framework -- we should always strive to make it easiest for the next developer who picks this up next, and this field comes with presumptions that won't hold true here. More importantly though, is that this field can be stomped on if the developer isn't careful: the framework has certain actions will implicitly invoke PersistableBusinessObjectBase#generateAndSetObjectIdIfNeeded(), which could potentially be a pernicious bug. Lastly, if it's necessary to take this to a higher level of optimization, this field will probably have to be moved out of the way, otherwise we'd be fighting the framework over it to make sure we're not taking any missteps.